server/routerlicious/packages/lambdas/src/deli/lambdaFactory.ts

```diff
@@ -57,25 +61,37 @@ export class DeliLambdaFactory extends EventEmitter implements IPartitionLambdaF
 
     public async create(config: IPartitionLambdaConfig, context: IContext): Promise<IPartitionLambda> {
         const { documentId, tenantId, leaderEpoch } = config;
+        const sessionMetric = createSessionMetric(tenantId, documentId, false, this.serviceConfiguration);
+        const sessionStartMetric = createSessionMetric(tenantId, documentId, true, this.serviceConfiguration);
+        const messageMetaData = {
+            documentId,
+            tenantId,
+        };
 
-        const tenant = await this.tenantManager.getTenant(tenantId, documentId);
-        const gitManager = tenant.gitManager;
+        let gitManager: IGitManager;
+        let dbObject: IDocument;
+
+        try {
+            const tenant = await this.tenantManager.getTenant(tenantId, documentId);
+            gitManager = tenant.gitManager;
+
+            // Lookup the last sequence number stored
+            // TODO - is this storage specific to the orderer in place? Or can I generalize the output context?
+            dbObject = await this.collection.findOne({ documentId, tenantId });
+        } catch (error) {
+            const errMsg = "Deli lambda creation failed";
+            context.log?.error(`${errMsg}. Exception: ${inspect(error)}`, { messageMetaData });
+            this.logSessionFailureMetrics(sessionMetric, sessionStartMetric, errMsg);
```

Comment:
If the lambda creation failed, do we consider that the session itself failed, or that only the start of the session failed? If I'm not mistaken, we have the possible combination of events:
1. SessionStart succeeds, Session succeeds
2. SessionStart succeeds, Session fails
3. SessionStart fails, Session fails
4. SessionStart fails, Session succeeds -> is this one possible?

If 4 is not a possible combination, then does it make sense to say that if StartSession fails, we know for sure that the Session failed? And if that is the case, does it make sense to have the concept of "session" if startSession failed before we could create such session?

Reply:
That is a good question. Right now, we are defining startsession success as a case where nothing in the session start went wrong ie all the lambdas are successfully up. We are tying the concept of session end to the closing of the deli lambda. There is still some more plumbing to do here. For instance, we are not considering the corrupted docs as failed sessions. I plan to do that in the next PR. Once that is done, the definitions will become more coherent.